File: objc/objcfind/objcfind_tests/FindFileTests.m
#import <XCTest/XCTest.h>
#import "FileTypes.h"
#import "FindFile.h"

@interface FindFileTests : XCTestCase
//@property FileTypes *fileTypes;
@end

@implementation FindFileTests

- (void)testFindFileAbsPath {
    NSString *path = @"/Users/cary/src/xfind/objc/objcfind/objcfind/FindFile.m";
    FindFile *findFile = [[FindFile alloc]
                              initWithFilePath:path
                              fileType:FileTypeCode];
    NSString *findFileString = [findFile description];
    XCTAssert([findFileString isEqualToString:path]);
}

- (void)testFindFileTildePath {
    NSString *path = @"~/src/xfind/objc/objcfind/objcfind/FindFile.m";
    FindFile *findFile = [[FindFile alloc]
                              initWithFilePath:path
                              fileType:FileTypeCode];
    NSString *findFileString = [findFile description];
    XCTAssert([findFileString isEqualToString:path]);
}

- (void)testFindFileRelPath1 {
    NSString *path = @"./FindFile.m";
    FindFile *findFile = [[FindFile alloc]
                              initWithFilePath:path
                              fileType:FileTypeCode];
    NSString *findFileString = [findFile description];
    XCTAssert([findFileString isEqualToString:path]);
}

- (void)testFindFileRelPath2 {
    NSString *path = @"../FindFile.m";
    FindFile *findFile = [[FindFile alloc]
                              initWithFilePath:path
                              fileType:FileTypeCode];
    NSString *findFileString = [findFile description];
    XCTAssert([findFileString isEqualToString:path]);
}

@end

File: swift/swiftfind/Tests/swiftfindTests/FindFileTests.swift
//
//  swiftfindTests.swift
//  swiftfindTests
//
//  Created by Cary Clark on 5/18/15.
//  Copyright (c) 2015 Cary Clark. All rights reserved.
//

import XCTest

import swiftfind

class FindFileTests: XCTestCase {

    func testFindFileAbsPath() {
        let path = "\(NSHomeDirectory())/src/xfind/swift/swiftfind/Sources/swiftfind/FindFile.swift"
        let findFile = FindFile(filePath: path, fileType: FileType.code)
        XCTAssertEqual(path, findFile.description)
    }

    func testFindFileTildePath() {
        let path = "~/src/xfind/swift/swiftfind/Sources/swiftfind/FindFile.swift"
        let findFile = FindFile(filePath: path, fileType: FileType.code)
        XCTAssertEqual(path, findFile.description)
    }

    func testFindFileRelPath1() {
        let path = "./FindFile.swift"
        let findFile = FindFile(filePath: path, fileType: FileType.code)
        XCTAssertEqual(path, findFile.description)
    }

    func testFindFileRelPath2() {
        let path = "../FindFile.swift"
        let findFile = FindFile(filePath: path, fileType: FileType.code)
        XCTAssertEqual(path, findFile.description)
    }

    static var allTests = [
        ("testFindFileAbsPath", testFindFileAbsPath),
        ("testFindFileTildePath", testFindFileTildePath),
        ("testFindFileRelPath1", testFindFileRelPath1),
        ("testFindFileRelPath2", testFindFileRelPath2),
    ]
}
